src/library/spi_library.jl

```diff
@@ -0,0 +1,119 @@
+function add_static_outage_data!(sys::PSY.System, gen_for_data::DataFrame)
+    for row in DataFrames.eachrow(gen_for_data)
+        λ, μ = SPI.rate_to_probability(row.FOR, row["MTTR Hr"])
+        transition_data = PSY.GeometricDistributionForcedOutage(;
+            mean_time_to_recovery = row["MTTR Hr"],
+            outage_transition_probability = λ,
+        )
+        comp = PSY.get_component(PSY.Generator, sys, row["GEN UID"])
+
+        if !isnothing(comp)
+            PSY.add_supplemental_attribute!(sys, comp, transition_data)
+            @debug "Added outage data supplemental attribute to $(row["GEN UID"]) generator"
+        else
+            @warn "$(row["GEN UID"]) generator doesn't exist in the System."
+        end
+    end
+end
+
+function add_timeseries_outage_data!(sys::PSY.System, rts_outage_ts_data::DataFrame)
+    # Time series timestamps
+    static_ts_summary = PSY.get_static_time_series_summary_table(sys)
+    s2p_meta = SPI.S2P_metadata(static_ts_summary)
+    step = s2p_meta.pras_resolution(s2p_meta.pras_timestep)
+    finish_datetime =
+        s2p_meta.first_timestamp + s2p_meta.pras_resolution((s2p_meta.N - 1) * step)
+    ts_timestamps = collect(StepRange(s2p_meta.first_timestamp, step, finish_datetime))
+
+    # Add λ and μ time series 
+    for row in DataFrames.eachrow(rts_outage_ts_data)
+        comp = PSY.get_component(PSY.Generator, sys, row.Unit)
+        λ_vals = Float64[]
+        μ_vals = Float64[]
+        for i in range(0; length = 12)
+            next_timestamp = s2p_meta.first_timestamp + Dates.Month(i)
+            λ, μ = SPI.rate_to_probability(row[3 + i], 48) # Assuming MTTR is 48
```

Comment:
We really shouldn't be importing SPI just for this one function.

Reply:
Done. I don't use SPI anymore.